desktop/main/NodeConfig.ts

```diff
@@ -21,19 +23,89 @@ export const loadNodeConfig = async (): Promise<NodeConfig> =>
         )
     : {};
 
+const getCustomNodeConfigName = (name: string) => `node-config.${name}.json`;
+
+export const writeCustomNodeConfig = async (
+  netName: string,
+  config: string
+): Promise<NodeConfig> => {
+  const netConfigPath = path.join(
+    USERDATA_DIR,
+    getCustomNodeConfigName(netName)
+  );
+
+  await fs.writeFile(netConfigPath, config, {
+    encoding: 'utf8',
+  });
+
+  return JSON.parse(config);
+};
+
+export const loadOrCreateCustomConfig = async (
+  netName: string
+): Promise<NodeConfig> =>
+  existsSync(path.join(USERDATA_DIR, getCustomNodeConfigName(netName)))
+    ? fs
+        .readFile(path.join(USERDATA_DIR, getCustomNodeConfigName(netName)), {
+          encoding: 'utf8',
+        })
+        .then((res) =>
+          res.startsWith('{') ? JSON.parse(res) : TOML.parse(res)
+        )
+    : writeCustomNodeConfig(netName, '{}');
+
 const loadSmeshingOpts = async (nodeConfig) => {
   const id = generateGenesisIDFromConfig(nodeConfig);
   const opts = StoreService.get(`smeshing.${id}`);
   return safeSmeshingOpts(opts, id);
 };
 
-export const downloadNodeConfig = async (networkConfigUrl: string) => {
+export const isNodeConfigAndDiscoveryConfigEqual = (
+  nodeConfig,
+  discoveryConfig
+) => {
+  let equal = true;
+  R.mergeDeepWithKey(
+    (_, clientConfig, config) => {
+      if (R.equals(clientConfig, config)) {
+        return clientConfig;
+      }
+
+      equal = false;
+
+      return config;
+    },
+    nodeConfig,
+    discoveryConfig
+  );
+
+  return equal;
+};
+
+export const downloadNodeConfig = async (
+  netName: string,
+  networkConfigUrl: string
+) => {
   const nodeConfig = await fetchNodeConfig(networkConfigUrl);
-  // Copy smeshing opts from previous node config or replace it with empty one
-  nodeConfig.smeshing = await loadSmeshingOpts(nodeConfig);
+  const clientConfig = await loadNodeConfig();
+
+  // download and write config only if not exists
+  if (!R.isEmpty(clientConfig)) {
+    // keep initial smeshing options
+    nodeConfig.smeshing = await loadSmeshingOpts(nodeConfig);
+    // forward config to $nodeConfig.next
+    return nodeConfig;
+  }
+
+  const customNodeConfig = await loadOrCreateCustomConfig(netName);
+  const config = R.isEmpty(customNodeConfig)
+    ? nodeConfig
+    : R.mergeDeepLeft(customNodeConfig, nodeConfig);
+
+  config.smeshing = await loadSmeshingOpts(config);
```

Comment:
We won't need storing smeshing opts in StoreService anymore, and it should not overwrite values stored in the custom config!

So, first of all, removing this line
```suggestion
```

Then, we will need to tweak a couple of places in SmesherManager to not store it in StoreService, but store it in custom config instead.

And bonus point, write a tiny migration function (which should be easily removed in the future), which will check if StoreService contains `smeshing` opts for the network and move it into custom config instead.